src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -0,0 +1,257 @@
+
+	parameters, err := parseParameter(r, vars)
+	if err != nil {
+		h.logger.Error("Bad Request", err, lager.Data{"appId": appId})
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad Request",
+			Message: err.Error(),
+		})
+		return
+	}
+	if metricType == "" {
+		h.logger.Error("Bad Request", nil, lager.Data{"appId": appId})
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad Request",
+			Message: "metrictype is required",
+		})
+		return
+	}
+	if instanceIndex != "" {
+		parameters.Add("instance-index", instanceIndex)
+	}
+
+	path, _ := routes.MetricsCollectorRoutes().Get(routes.GetMetricHistoriesRouteName).URLPath("appid", appId, "metrictype", metricType)
+
+	url := h.conf.MetricsCollector.MetricsCollectorUrl + path.RequestURI() + "?" + parameters.Encode()
+
+	resp, err := h.metricsCollectorClient.Get(url)
+	if err != nil {
+		h.logger.Error("Failed to retrieve metrics history from metricscollector", err, lager.Data{"url": url})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error retrieving metrics history from metricscollector"})
+		return
+	}
+	defer resp.Body.Close()
+
+	responseData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		h.logger.Error("Error occured during parsing metrics histories result", err, lager.Data{"url": url})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error parsing metric history from metricscollector"})
+		return
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		h.logger.Error("Error occured during getting metric histories", nil, lager.Data{"statusCode": resp.StatusCode, "body": string(responseData)})
+		handlers.WriteJSONResponse(w, resp.StatusCode, models.ErrorResponse{
+			Code:    string(resp.StatusCode),
+			Message: string(responseData)})
+		return
+	}
+
+	paginatedResponse, err := paginateResource(responseData, parameters, r)
+	if err != nil {
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: err.Error()})
+	}
+
+	handlers.WriteJSONResponse(w, resp.StatusCode, paginatedResponse)
+}
+
+func (h *PublicApiHandler) GetApiInfo(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	info, err := ioutil.ReadFile(h.conf.InfoFilePath)
+	if err != nil {
+		h.logger.Error("Failed to info file", err, lager.Data{"info-file-path": h.conf.InfoFilePath})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Failed to load info"})
+		return
+	}
+	w.Write([]byte(info))
+}
+
+func (h *PublicApiHandler) GetHealth(w http.ResponseWriter, r *http.Request, vars map[string]string) {
```

Comment:
looks like we always return "alive" then why we need it?

Reply:
I would assume this should be prometheus.

Reply:
it is a legacy design issue.  Before moving to Prometheus,   there is a /health endpoint on api server to show uptime.  Also  autoscaler CLI is using the /health endpoint to determine whether the api server is accessible.  i.e.  CLI expects to get 200 OK with /health
https://github.com/cloudfoundry-incubator/app-autoscaler-cli-plugin/blob/master/src/cli/api/apihelper.go#L140